src/Orleans.Core/Logging/ILoggerExtensions.cs

```diff
@@ -252,7 +252,7 @@ internal static void Assert(this ILogger logger, ErrorCode errorCode, bool condi
             logger.Fail(errorCode, "Assert failed with message = " + message);
         }
 
-        internal static void Fail(this ILogger logger, ErrorCode errorCode, string message)
+        public static void Fail(this ILogger logger, ErrorCode errorCode, string message)
```

Comment:
In time we should attempt to remove our dependence on these

Reply:
I'm fine with leaving it as is.

Reply:
The main impetus is for semantic logging, since that lets our users perform richer log searches